File: C#/Mapa/Mapa.cs
using System;
using System.Collections.Generic;

namespace Mapa
{
    //Nombre de la clase mapa
    public class Mapa
    {
        //Instancia de un mapa 
        private Dictionary<string, string> mapa;

        //Métodos de la clase
        public Mapa()
        {
            mapa = new Dictionary<string, string>();
        }

        public void agregar(string palabraUsada, string significado)
        {
            mapa.Add(palabraUsada, significado);
        }

        public string buscar(string palabraUsada)
        {
            return mapa[palabraUsada];
        }

        public void ordenar()
        {
            var valueDato = new List<string>(mapa.Keys);
            valueDato.Sort();

            var nuevoMapa = new Dictionary<string, string>();
            foreach (var valor in valueDato)
            {
                nuevoMapa.Add(valor, mapa[valor]);
            }
            mapa = nuevoMapa;
        }

        public void desordenar()
        {
            var valueDato = new List<string>(mapa.Keys);

            valueDato.Reverse();
            var nuevoMapa = new Dictionary<string, string>();

            foreach (var valor in valueDato)
            {
                nuevoMapa.Add(valor, mapa[valor]);
            }
            mapa = nuevoMapa;
        }

        public void eliminar(string palabraUsada)
        {

            mapa.Remove(palabraUsada);
        }

        public void insertar(string palabraUsada, string significado)
        {

            mapa.Add(palabraUsada, significado);
        }

        public void recorridoDeinicioAFin()
        {

            foreach (var palabraUsada in mapa)
            {
                Console.WriteLine(palabraUsada.Key+": "+palabraUsada.Value);
            }
        }

        public void recorridoDeFinAInicio()
        {
           
            Dictionary<string, string> ayuda = new Dictionary<string, string>(mapa);
            ayuda = mapa;

            var valueDato = new List<string>(ayuda.Keys);

            valueDato.Reverse();
            
            foreach (var valor in valueDato)
            {
                Console.WriteLine(valor+"----> "+ayuda[valor]);
            }

        }

        //Método para imprimir
        public void imprimir()
        {
            Console.WriteLine();
           
            Console.WriteLine("*******SIGNIFICADOS DE PALABRAS EN IDIOMAS*******");
           
            foreach (var palabraUsada in mapa)
            {
                Console.WriteLine(palabraUsada.Key+": "+palabraUsada.Value);
            }
           
                      
            Console.WriteLine();
        }
    }
}

File: PYTHON/Mapa/Mapa.py
import collections

#Nombre de la clase
class Mapa :
    mapa = {}
    #Métodos de la clase
    def agregar(self, palabra, significado):
        self.mapa[palabra] = significado
    
    def buscar(self, palabra):
        return self.mapa[palabra]
    
    def ordenar(self):
        self.mapa = collections.OrderedDict(sorted(self.mapa.items()))
    
    def desordenar(self):
       self.mapa = collections.OrderedDict(sorted(self.mapa.items(), reverse=True))

    def eliminar(self, palabra):
        del self.mapa[palabra]

    def insertar(self, palabra, significado):
        self.mapa[palabra] = significado

    def recorridoDeInicioAFin(self):
        for key in self.mapa:
            print(key+': '+self.mapa[key])
    

    def recorridoDeFinAInicio(self):
        mapaDeFinAInicio = collections.OrderedDict(sorted(self.mapa.items()))
        for key in mapaDeFinAInicio:
            print(key+'---> '+mapaDeFinAInicio[key])

    def imprimir(self):
        print("*******SIGNIFICADOS DE PALABRAS EN IDIOMAS*********")
        for key in self.mapa:
            print(key+': '+self.mapa[key])
